### ISSUE TITLE
'max_redirect' doesn't work

### ISSUE THREAD

**Issue by tnt-dev** [benoitc/hackney#170]
Example:

```
hackney:get("http://rdrct.herokuapp.com/1", [], "", [{follow_redirect, false}]).
{ok,301,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Thu, 26 Feb 2015 16:08:57 GMT">>},
     {<<"Content-Length">>,<<"0">>},
     {<<"Location">>,<<"/2">>},
     {<<"Via">>,<<"1.1 vegur">>}],
    #Ref<0.0.0.1164>}
```
```
hackney:get("http://rdrct.herokuapp.com/2", [], "", [{follow_redirect, false}]).
{ok,301,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Thu, 26 Feb 2015 16:11:24 GMT">>},
     {<<"Content-Length">>,<<"0">>},
     {<<"Location">>,<<"/3">>},
     {<<"Via">>,<<"1.1 vegur">>}],
    #Ref<0.0.0.1299>}
```
```
hackney:get("http://rdrct.herokuapp.com/3", [], "", []).
{ok,200,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Thu, 26 Feb 2015 16:17:57 GMT">>},
     {<<"Content-Length">>,<<"2">>},
     {<<"Via">>,<<"1.1 vegur">>}],
    #Ref<0.0.0.1695>}
```
```
hackney:get("http://rdrct.herokuapp.com/1", [], "", [{follow_redirect, true}, {max_redirect, 1}]).
{ok,200,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Thu, 26 Feb 2015 16:18:49 GMT">>},
     {<<"Content-Length">>,<<"2">>},
     {<<"Via">>,<<"1.1 vegur">>}],
    #Ref<0.0.0.1753>}
```

Last request must return {max_redirect_overflow} error.

**Comment by benoitc** [comment]
Thanks for the test! I just found the issue. I will make a patch tomorrow since it starts to be late :) Os this related to #168?

**Comment by tnt-dev** [comment]
Yes, this issie is related to #168 :)

The problem is that this: https://github.com/benoitc/hackney/blob/master/src/hackney_client/hackney.erl#L803-L819 actually doesn't work and we make new reqeust with enabled request support. This breaks planned behaviour and leads to problems. When it will be fixed you will see that problem in #168 is real :)

**Comment by tnt-dev** [comment]
Sorry, but 1.1.0 version returns the same result:
```
hackney:get("http://rdrct.herokuapp.com/1", [], "", [{follow_redirect, true}, {max_redirect, 1}]).
{ok,200,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Thu, 19 Mar 2015 12:20:14 GMT">>},
     {<<"Content-Length">>,<<"2">>},
     {<<"Via">>,<<"1.1 vegur">>}],
    #Ref<0.0.0.6604>}
```
```
hackney:head("http://rdrct.herokuapp.com/1", [], "", [{follow_redirect, true}, {max_redirect, 1}]).
{ok,200,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Thu, 19 Mar 2015 12:21:07 GMT">>},
     {<<"Content-Length">>,<<"2">>},
     {<<"Via">>,<<"1.1 vegur">>}]}
```
